src/utilities/decorators/withViewport.tsx

```diff
@@ -61,6 +61,13 @@ export function withViewport<P, S>(ComposedComponent: any): any {
       );
     }
 
+    /**
+     * Accessor for the instance of the component being wrapped by withViewport.
+     */
+    public get composedComponentInstance() {
```

Comment:
I don't think this is very intuitive. Consider this. I write a DetailsList, I add a focus method. Consumers love this, ok now we can tell the detailslist to focus.

```
this.refs.detailsList.focus();
```

The detailslist needs width to measure, so it decides to use withViewport:
```
@withViewport
class DetailsList ... { ... }
```

Now the consumer's code has been broken, because detailsList.focus() is no longer accessible. If I understand correctly, they would have to change their code:

this.refs.detailsList.composedComponentInstance.focus();

This is really unintuitive. We are basically "leaking" the decorator implementation details to the consumer.

Maybe better approach would be to actually hoist the methods. Some pseudocode example:

```typescript
function hoistMethods(destination, source, exclusions) {
  for (let methodName in source) {
     if (
       typeof source[methodName] === 'function' && 
       destination[methodName] === undefined &&
       exclusions.indexOf(methodName) === -1) {
      destination[methodName] = () => source[methodName].apply(source, arguments);
    }
  }
}
```